### PR TITLE
mobile_image_mounter: add many changes for ios16

### PR DESCRIPTION
**Pull request by doronz88**

### CODE REVIEW

File: pymobiledevice3/services/mobile_image_mounter.py [hunk 1]
```diff
@@ -43,13 +43,17 @@ def is_image_mounted(self, image_type: str) -> bool:
...
-    def umount(self, image_type, mount_path, signature):
+    def umount(self, mount_path: str, image_type: str = None, signature: bytes = None) -> None:
         """ umount image. """
-        self.service.send_plist({'Command': 'UnmountImage',
-                                 'ImageType': image_type,
-                                 'MountPath': mount_path,
-                                 'ImageSignature': signature})
-        response = self.service.recv_plist()
+        request = {'Command': 'UnmountImage', 'MountPath': mount_path}
+        response = self.service.send_recv_plist(request)
```
Comment (matan1008) [inline pymobiledevice3/services/mobile_image_mounter.py]: Move down to line 56

File: pymobiledevice3/services/mobile_image_mounter.py [hunk 2]
```diff
@@ -59,26 +63,40 @@ def umount(self, image_type, mount_path, signature):
...
+        if image_type == 'Cryptex':
+            if trust_cache is None:
+                raise ValueError('Cryptex image requires a ImageTrustCache to be also supplied')
 
-        if 'Developer mode is not enabled' in result.get('DetailedError', ''):
+            if info_plist is None:
+                raise ValueError('Cryptex image requires a ImageInfoPlist to be also supplied')
+
+        if trust_cache is not None:
+            request['ImageTrustCache'] = trust_cache
```
Comment (matan1008) [inline pymobiledevice3/services/mobile_image_mounter.py:84]: Is it exclusive to cryptexes?

File: pymobiledevice3/services/mobile_image_mounter.py (hunk 1 above)
Comment (doronz88) [reply pymobiledevice3/services/mobile_image_mounter.py]: done

File: pymobiledevice3/services/mobile_image_mounter.py (hunk 2 above)
Comment (doronz88) [reply pymobiledevice3/services/mobile_image_mounter.py:84]: yeah. verified also on sever side

Comment (doronz88) [reply pymobiledevice3/services/mobile_image_mounter.py:84]: refactored

**Pull request merged**